File: articles/coordination-problems-in-institutional-systems/fortran/coordination_score.f90
program coordination_score
  implicit none

  real :: trust, information_quality, communication_clarity, focal_salience
  real :: authority_signal, norm_strength, learning_capacity, uncertainty
  real :: adaptation_burden, competing_standards, competing_authority
  real :: distributional_attention, score

  trust = 80.0
  information_quality = 78.0
  communication_clarity = 82.0
  focal_salience = 76.0
  authority_signal = 74.0
  norm_strength = 70.0
  learning_capacity = 72.0
  uncertainty = 30.0
  adaptation_burden = 35.0
  competing_standards = 25.0
  competing_authority = 20.0
  distributional_attention = 68.0

  score = 0.14 * trust + &
          0.14 * information_quality + &
          0.13 * communication_clarity + &
          0.12 * focal_salience + &
          0.10 * authority_signal + &
          0.10 * norm_strength + &
          0.09 * learning_capacity - &
          0.13 * uncertainty - &
          0.07 * adaptation_burden - &
          0.06 * competing_standards - &
          0.05 * competing_authority + &
          0.04 * distributional_attention

  print *, "Institutional coordination raw score:", score
end program coordination_score

File: articles/coordination-problems-in-institutional-systems/go/coordination_score.go
package main

import "fmt"

type CoordinationCase struct {
	Trust                 float64
	InformationQuality    float64
	CommunicationClarity  float64
	FocalSalience         float64
	AuthoritySignal       float64
	NormStrength          float64
	LearningCapacity      float64
	Uncertainty           float64
	AdaptationBurden      float64
	CompetingStandards    float64
	CompetingAuthority    float64
	DistributionalAttention float64
}

func CoordinationScoreRaw(x CoordinationCase) float64 {
	return 0.14*x.Trust +
		0.14*x.InformationQuality +
		0.13*x.CommunicationClarity +
		0.12*x.FocalSalience +
		0.10*x.AuthoritySignal +
		0.10*x.NormStrength +
		0.09*x.LearningCapacity -
		0.13*x.Uncertainty -
		0.07*x.AdaptationBurden -
		0.06*x.CompetingStandards -
		0.05*x.CompetingAuthority +
		0.04*x.DistributionalAttention
}

func main() {
	demo := CoordinationCase{80, 78, 82, 76, 74, 70, 72, 30, 35, 25, 20, 68}
	fmt.Printf("Institutional coordination raw score: %.2f\n", CoordinationScoreRaw(demo))
}
